synapse/storage/search.py

```diff
@@ -334,7 +335,8 @@ def store_search_entries_txn(self, txn, entries):
                     entry.event_id,
                     entry.room_id,
                     entry.key,
-                    entry.value,
+                    # Only store words while discarding special characters
+                    ' '.join(re.findall(r"([\w\-]+)", entry.value, re.UNICODE)),
```

Comment:
1. what exactly is this doing? I suggest writing a longer comment outside the `args = ` expression, giving some examples and explanation.

2. `re.UNICODE` is obsolete

3. Any reason we use `re.findall` rather than `re.finditer`?

Reply:
Does the same thing as is done here: https://github.com/matrix-org/synapse/blob/develop/synapse/storage/search.py#L695

No particular reason, `finditer` should work too.

I'll add some comments with explanations.

Reply:
Not sure what to do about the UNICODE thing, can we just remove it?